ebean-test/src/test/java/org/tests/idkeys/TestSequenceMultiTenant.java

```diff
@@ -66,32 +68,100 @@ public DataSource dataSource(Object tenantId) {
         if (tenantId == null) {
           tenantId = "1";
         }
-        return map.computeIfAbsent(tenantId,
-            TestSequenceMultiTenant::createDataSource);
+        return map.computeIfAbsent(tenantId, this::createDataSource);
+      }
+
+      private DataSource createDataSource(Object tenantId) {
+
+        DatabaseConfig config = new DatabaseConfig();
+
+        config.setName("h2multitenantseq");
+        config.loadFromProperties();
+        config.setDdlRun(true);
+        config.setDdlExtra(false);
+        config.setRegister(false);
+        config.setDefaultServer(false);
+        config.getDataSourceConfig().setUrl("jdbc:h2:mem:h2multitenantseq-" + tenantId);
+
+        return DatabaseFactory.create(config).pluginApi().dataSource();
       }
     });
 
     config.getClasses().add(GenKeySequence.class);
 
-    return EbeanServerFactory.create(config);
+    return DatabaseFactory.create(config);
+  }
+
+  
+  /**
+   * Tests sequences using multi tenancy per schema
+   */
+  @Test
+  public void test_multi_tenant_schema_sequences() throws SQLException {
+    createDDl("PUBLIC");
+    createDDl("TENANT_SCHEMA_1");
+    createDDl("TENANT_SCHEMA_2");
+
+    Database db = setupSchema();
+
+    UserContext.set("4711", "1");
+    assertEquals(1L, db.nextId(GenKeySequence.class));
+    assertEquals(2L, db.nextId(GenKeySequence.class));
+
+    UserContext.set("5711", "2");
+    assertEquals(1L, db.nextId(GenKeySequence.class));
+
+    UserContext.set("4711", "1");
+    assertEquals(3L, db.nextId(GenKeySequence.class));
   }
 
-  private static DataSource createDataSource(Object tenantId) {
+  private Database setupSchema() {
```

Comment:
With the support of @rPraml we were able to write a failing test for tenantmode schema. THe main question seems to be, were the sequence schould live.